packages/storybook/src/executors/utils.ts

```diff
@@ -216,6 +224,58 @@ export function resolveCommonStorybookOptionMapper(
 
       buildProjectName = defaultProjectName;
 
+      if (defaultProjectName) {
+        logger.warn(
+          `
+          No projectBuildConfig was provided. Nx will use the defaultProject as set in your workspace configuration
+          to build Storybook. This can lead to compilation errors for Storybook.
```

Comment:
I'm not sure we should fallback to running Storybook with the default project resolution at all because we already know it's not gonna work.

I would stop the run at this point and mention how to fix it as you do below.